moderation_test.go

```diff
@@ -27,6 +27,18 @@ func TestModerations(t *testing.T) {
 	checks.NoError(t, err, "Moderation error")
 }
 
+// TestModerationsWithIncorrectModel Tests passing an incorrect model to Moderations request.
+func TestModerationsWithIncorrectModel(t *testing.T) {
+	client, server, teardown := setupOpenAITestServer()
+	defer teardown()
+	server.RegisterHandler("/v1/moderations", handleModerationEndpoint)
+	_, err := client.Moderations(context.Background(), ModerationRequest{
+		Model: GPT3Dot5Turbo,
+		Input: "I want to kill them.",
+	})
+	checks.ErrorIs(t, err, ErrModerationInvalidModel)
+}
```

Comment:
@MunaerYesiyan Thanks for PR! How about trying to write exhaustive tests for both valid and invalid models in table driven tests?

Reply:
https://github.com/sashabaranov/go-openai/pull/437/commits/e1ae0560840e0df535a9fe27b790801280bdefe6